pkg/melange/testdata/duplicates_dir/foo2.yaml

```diff
@@ -0,0 +1,3 @@
+package:
+  name: foo
```

Comment:
I think we might also want to fail if `.package.name` is not the same as the filename.

These are potentially different/separate issues, but it does I think violate some assumptions we've made, and is worth avoiding.

If we have that check, we also get uniqueness "for free" since there can't be two `foo.yaml`s

Reply:
If the .package.name field must match the filename, couldn't you make an argument that the field is redundant and should be removed?